### ISSUE TITLE
systemd-networkd: Can't set dhcp hostname

### ISSUE THREAD

**Issue by max06** [NixOS/nixpkgs#364198]
## Describe the bug

Systemd-networkd fails setting the dhcp-acquired hostname due to `Access denied`.

## Steps To Reproduce

1. Either don't define `networking.hostName` (defaults to `nixos`; not documented) or use `networking.hostName = ""`.
2. Enable `networking.useDHCP` and `networking.useNetworkd`
3. Check `journalctl -u systemd-networkd` 

## Actual behaviour

```
Dec 11 11:06:14 nixos systemd[1]: Started Network Configuration.
Dec 11 11:06:14 nixos systemd-networkd[53495]: ens18: Configuring with /etc/systemd/network/99-ethernet-default-dhcp.netw>
Dec 11 11:06:14 nixos systemd-networkd[53495]: ens18: DHCPv4 address 192.168.27.201/24, gateway 192.168.27.1 acquired fro>
Dec 11 11:06:14 nixos systemd-networkd[53495]: Could not set hostname: Access denied
```

## Expected behavior

The hostname should be set.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 25.05 (Warbler), 25.05pre-git`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `not found`
 - 
## Notify maintainers

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
> 1. defaults to `nixos`; not documented

It's documented:

https://search.nixos.org/options?channel=24.11&show=networking.hostName&from=0&size=50&sort=relevance&type=packages&query=networking.hostName

**Comment by max06** [comment]
@eclairevoyant technically correct - the option is documented. It's default points to `config.system.nixos.distroId` which is missing. And neither it says `nixos` anywhere.

**Comment by eclairevoyant** [comment]
You can always inspect the config via `nixos-option` (without flakes) or `nix eval FLAKE#nixosConfigurations.HOST.config.system.nixos.distroId` (with flakes)

**Comment by max06** [comment]
Yup, I followed along on a repl. I still think this value should be in the documentation as well. Anyways, this does not solve the original issue 😄
